dsn.go

```diff
@@ -65,8 +65,19 @@ type Config struct {
 	MultiStatements          bool // Allow multiple statements in one query
 	ParseTime                bool // Parse time values to time.Time
 	RejectReadOnly           bool // Reject read-only connections
+
+	Compress          CompressionMode // Compress packets
+	MinCompressLength int             // Don't compress packets less than this number of bytes
```

Comment:
I don't want to add public members because it makes keeping backward compatibility hard. We are suffered by some members already.

For example, MySQL deprecated the `compress` option in fever of `compression-algorithms`. So we may change these options in the future too.
https://dev.mysql.com/doc/refman/8.2/en/connection-options.html#option_general_compress

We need to discuss about new style to set options.